packages/typespec-azure-resource-manager/lib/common-types/network-security-perimeter.tsp

```diff
@@ -0,0 +1,269 @@
+  provisioningState?: NetworkSecurityPerimeterConfigurationProvisioningState;
+
+  /** List of provisioning issues, if any */
+  @visibility("read")
+  @OpenAPI.extension("x-ms-identifiers", [])
+  provisioningIssues?: ProvisioningIssue[];
+
+  networkSecurityPerimeter?: NetworkSecurityPerimeter;
+  resourceAssociation?: ResourceAssociation;
+  profile?: NetworkSecurityProfile;
+}
+
+/** Describes a provisioning issue for a network security perimeter configuration */
+@added(Versions.v5)
+model ProvisioningIssue {
+  /** Name of the issue */
+  @visibility("read")
+  name?: string;
+
+  @visibility("read")
+  properties?: ProvisioningIssueProperties;
+}
+
+/** Details of a provisioning issue for a network security perimeter (NSP) configuration. Resource providers should generate separate provisioning issue elements for each separate issue detected, and include a meaningful and distinctive description, as well as any appropriate suggestedResourceIds and suggestedAccessRules */
+@added(Versions.v5)
+model ProvisioningIssueProperties {
+  /** Type of issue */
+  @visibility("read")
+  issueType?: IssueType;
+
+  /** Severity of the issue. */
+  @visibility("read")
+  severity?: Severity;
+
+  /** Description of the issue */
+  @visibility("read")
+  description?: string;
+
+  /** Fully qualified resource IDs of suggested resources that can be associated to the network security perimeter (NSP) to remediate the issue. */
+  @visibility("read")
+  suggestedResourceIds?: Azure.Core.armResourceIdentifier[];
+
+  /** Access rules that can be added to the network security profile (NSP) to remediate the issue. */
+  @visibility("read")
+  @OpenAPI.extension("x-ms-identifiers", [])
+  suggestedAccessRules?: AccessRule[];
+}
+
+/** Information about resource association */
+@added(Versions.v5)
+model ResourceAssociation {
+  /** Name of the resource association */
+  name?: string;
+
+  accessMode?: ResourceAssociationAccessMode;
+}
+
+/** Network security perimeter configuration profile */
+@added(Versions.v5)
+model NetworkSecurityProfile {
+  /** Name of the profile */
+  name?: string;
+
+  /** Current access rules version */
+  accessRulesVersion?: int32;
+
+  /** List of Access Rules */
+  @OpenAPI.extension("x-ms-identifiers", ["name"])
+  accessRules?: AccessRule[];
+
+  /** Current diagnostic settings version */
+  diagnosticSettingsVersion?: int32;
+
+  /** List of log categories that are enabled */
+  enabledLogCategories?: string[];
+}
+
+/** Result of a list NSP (network security perimeter) configurations request. */
+@added(Versions.v5)
+model NetworkSecurityPerimeterConfigurationListResult {
+  /** Array of network security perimeter results. */
+  value?: NetworkSecurityPerimeterConfiguration[];
+
+  /** The link used to get the next page of results. */
+  nextLink?: url;
+}
+
+/**
+ * The default NetworkSecurityPerimeterConfigurationName.
+ */
+@added(Versions.v5)
+model NetworkSecurityPerimeterConfigurationNameParameter {
```

Comment:
We need to decide how to handle the segment name in this parameter model.  We might wanto to do this: https://github.com/Azure/typespec-azure/issues/1358

Or there may be a static value that is always be used.  This depends on usage inside specs

Reply:
Mark, I think that is the link for the bug that you mentioned above, could you please share the link for the suggestion?

Reply:
Yes, I meant a template, like this: https://github.com/Azure/typespec-azure/blob/95cc6b95355aac89280f39bc4b5b9ae94f6ae16d/packages/typespec-azure-resource-manager/lib/common-types/private-links.tsp#L117-L127